pkg/queue/bucketing_test.go

```diff
@@ -252,6 +253,87 @@ func TestDescendingRecord(t *testing.T) {
 	}
 }
 
+func TestRequestsSuddenStop(t *testing.T) {
+	RegisterTestingT(t)
+	now := time.Date(2024, 6, 26, 12, 0, 0, 0, time.UTC)
```

Comment:
using a fixed timestamp to get deterministic buckets, given the circular buffer uses modulo on seconds, this is one way to ensure the bucket values in the window buffer can be asserted against easily